dora.py

```
@@ -292,785 +248,461 @@ def can_process(dependency_update: components.ComponentVector):
-        )
-        for commit in component_dependency_change_with_commits.commits:
-            if (
-                    (
-                            commit_date := dateutil.parser.isoparse(commit.commit.author['date'])
-                    ) > (
-                    datetime.datetime.now(datetime.timezone.utc)
-                    - datetime.timedelta(days=time_span_days)
+        new_ref_component: ocm.Component = (await component_descriptor_lookup(
+            ocm.ComponentIdentity(
+                name=cnudie.util.to_component_name(target_update.referenced_component),
+                version=target_update.referenced_component_version_newer_release,
             )
-            ):
-                time_differences.append(deployment_date -  commit_date)
-
-    if not time_differences:
-        time_differences.append(datetime.timedelta(seconds=-1))
-    if calculation_type is CalculationType.MEDIAN:
-        result_in_seconds: float = statistics.median(
-            [time_difference.total_seconds()
-             for time_difference in time_differences]
-        )
-    else:
-        result_in_seconds: float = statistics.mean(
-            [time_difference.total_seconds()
-             for time_difference in time_differences]
-        )
-    return datetime.timedelta(seconds=result_in_seconds)
-
-
-@caching.cached(
-    cache=caching.LFUFilesystemCache(max_total_size_mib=128),
-    key_func=_cache_key_gen_dora,
-)
-def dora_changes_monthly(
-    component_dependency_changes_with_commits: list[
-        ComponentDependencyChangeWithCommits
-    ],
-    time_span_days: int,
-) -> list[DoraMonthlyResponse]:
+        )).component
 
-    code_changes_by_month: dict[
-        tuple[int, int],
-        list[tuple[datetime.datetime, CodeChange]],
-    ] = (
-        collections.defaultdict(list[tuple[datetime.datetime, CodeChange]])
-    )
+        if not old_ref_component or not new_ref_component:
+            print ("ERROR, bad input data")
```

please use logger instead; also, if this should be communicated to client -> raise `aiohttp.web.HTTP...` exception

---

+1 for using logger. + if it is actually an error, it should definitively not go by unnoticed (log output is equivalent to silently ignoring an error, considering we cannot expect delivery-service's log-output to actually be read during operations).

---

This is a left over from debugging :)

Currently I can not imagine a case where it will break. target_updates should only contain updates where both versions exist.